sdk/test/trace/tracer_test.cc

```diff
@@ -166,20 +171,25 @@ TEST(Tracer, StartSpanWithOptionsTime)
 
 TEST(Tracer, StartSpanWithAttributes)
 {
+
   std::shared_ptr<std::vector<std::unique_ptr<SpanData>>> spans_received(
       new std::vector<std::unique_ptr<SpanData>>);
   // The default tracer has empty sampling result attribute
   auto tracer = initTracer(spans_received);
 
   // Start a span with all supported scalar attribute types.
-  tracer->StartSpan("span 1", {{"attr1", "string"},
-                               {"attr2", false},
-                               {"attr1", 314159},
-                               {"attr3", (unsigned int)314159},
-                               {"attr4", (int64_t)-20},
-                               {"attr5", (uint64_t)20},
-                               {"attr6", 3.1},
-                               {"attr7", "string"}});
```

Comment:
Since you don't need `span_first` again later, can you just tack the ->End() onto the StartSpan call?

Reply:
Added to end.